### ISSUE TITLE
Figure out how to delay initiation until after ModularSensors completes start up mode.

### ISSUE THREAD

**Issue by aufdenkampe** [EnviroDIY/Trigger#4]
@bschulz1701 & @SRGDamia1 

Modular sensors collects and logs data every 2 minutes for the first 20 minutes, to help with diagnoses when a logger is deployed.  However, when I add the trigger function, it often hangs. It's getting hung up because there are not valid values from the Yosemitech sensors during the first few cycles of startup.

Do you have suggestions for how to delay the trigger function until after startup mode starts?

**Comment by bschulz1701** [comment]
Not sure if this is the most elegant solution, but in the main loop you could make a test condition with a flag to prevent trigger from testing unless it has been a certain amount of time, like 
```
logger.log();
static bool TestTrig = false;
if(millis() > 300000 || TestTrig == true) { //Only trigger/set flag after the first 5 minutes 
TestTrig = true; //Set flag
trip.test(...); //Call the trigger test function
}
```

This would require no messing with the existing modular sensors stuff and should prevent it from calling getValue() before things are initialized.

**Comment by aufdenkampe** [comment]
@bschulz1701, thanks for the quick response and that suggestion!

Will `millis()` reset when the system sleeps/wakes every 10 minutes? That would be a problem.

**Comment by aufdenkampe** [comment]
BTW, I had a typo in the intro paragraph. Startup mode lasts for **20** minutes, with measurements every 2 minutes.

**Comment by bschulz1701** [comment]
No, it should not, it will go back to 0 on a reset or when it rolls over (every like 15 days), but this is why we use a flag as well, so even when it rolls over it will still test the trigger function since the flag is set

**Comment by aufdenkampe** [comment]
@bschulz1701, that's awesome, and a great solution! I'll implement it and push the example back to the Trigger repo.

**Comment by SRGDamia1** [comment]
Why aren't there valid values during the initial 2 minute logging intervals?

**Comment by aufdenkampe** [comment]
@SRGDamia1, good question.  The sonde takes about a minute to fully warm up before giving Conductivity values, but it begins responding with good temperature and DO values nearly immediately.
